lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -1059,7 +1059,13 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     
     auto objectTy = value->getType().unwrapOptionalType();
     
-    require(objectTy.isReferenceCounted(F.getModule()),
+    // C++ foreign reference types are represented as trivially lowered types
+    // since they do not require retain/release calls.
+    bool isImmortalFRT = objectTy.isForeignReferenceType() &&
```

Comment:
Is this also true for `SWIFT_UNSAFE_REFERENCE`s?

Reply:
Yeap! From the compiler point of view, they're the same thing, except if strictly safe mode is enabled – both `SWIFT_UNSAFE_REFERENCE` and `SWIFT_IMMORTAL_REFERENCE` macros expand into `retain:immortal` and `release:immortal`.